homeassistant/components/homematic.py

```diff
@@ -0,0 +1,528 @@
+                            component.DOMAIN,
+                            config)
+
+                        # Fire discovery event
+                        _HM_DISCOVER_HASS.bus.fire(
+                            EVENT_PLATFORM_DISCOVERED, {
+                                ATTR_SERVICE: discovery_type,
+                                ATTR_DISCOVERED: {
+                                    ATTR_DISCOVER_DEVICES:
+                                    found_devices,
+                                    ATTR_DISCOVER_CONFIG: ''
+                                    }
+                                }
+                            )
+                    # pylint: disable=broad-except
+                    except Exception as err:
+                        _LOGGER.error("Failed to autotetect %s with" +
+                                      "error '%s'", component_name, str(err))
+            for dev in devices_not_created:
+                if dev in HOMEMATIC_DEVICES:
+                    try:
+                        for hm_element in HOMEMATIC_DEVICES[dev]:
+                            hm_element.link_homematic()
+                            # Need to wait, if you have a lot devices we don't
+                            # to overload CCU/Homegear
+                            time.sleep(1)
+                    # pylint: disable=broad-except
+                    except Exception as err:
+                        _LOGGER.error("Failed link %s with" +
+                                      "error '%s'", dev, str(err))
+
+
+def _get_devices(device_type, keys):
+    """Get devices."""
+    from homeassistant.components.binary_sensor.homematic import \
+        SUPPORT_HM_EVENT_AS_BINMOD
+
+    # run
+    device_arr = []
+    if not keys:
+        keys = HOMEMATIC.devices
+    for key in keys:
+        device = HOMEMATIC.devices[key]
+        if device.__class__.__name__ in HM_DEVICE_TYPES[device_type]:
+            elements = device.ELEMENT + 1
+            metadata = {}
+
+            # Load metadata if needed to generate a param list
+            if device_type is DISCOVER_SENSORS:
+                metadata.update(device.SENSORNODE)
+            elif device_type is DISCOVER_BINARY_SENSORS:
+                metadata.update(device.BINARYNODE)
+
+                # Also add supported events as binary type
+                for event, channel in device.EVENTNODE.items():
+                    if event in SUPPORT_HM_EVENT_AS_BINMOD:
+                        metadata.update({event: channel})
+
+            params = _create_params_list(device, metadata)
+
+            # Generate options for 1...n elements with 1...n params
+            for channel in range(1, elements):
+                for param in params[channel]:
+                    name = _create_ha_name(name=device.NAME,
+                                           channel=channel,
+                                           param=param)
+                    ordered_device_dict = OrderedDict()
```

Comment:
Any reason you're using an `OrderedDict` here instead of a dict declaration?

```python
device_dict = {
  "platform": "homematic",
  "address": key,
  …
}
```